client/src/modules/stock/movements/registry.js

```diff
@@ -210,32 +254,44 @@ function StockMovementsController(Stock, Notify,
       // force expand grid
       vm.grouping.unfoldAllGroups();
     })
-    .catch(Notify.handleError);
+    .catch(Notify.handleError)
+    .finally(function () {
+      vm.loading = false;
+    });
   }
 
   // search modal
   function search() {
-    Modal.openSearchMovements()
-    .then(function (filters) {
-      if (!filters) { return; }
+    var filtersSnapshot = Stock.filter[filterKey].formatHTTP();
 
-      vm.isToday = false;
-      reload(filters);
-    })
-    .catch(Notify.handleError);
-  }
+    Modal.openSearchMovements(filtersSnapshot)
+      .then(function (changes) {
+        Stock.filter[filterKey].replaceFilters(changes);
+        Stock.cacheFilters(filterKey);
+        vm.latestViewFilters = Stock.filter[filterKey].formatView();
 
-  // reload
-  function reload(filters) {
-    vm.filters = filters;
-    vm.formatedFilters = SearchFilterFormat.formatDisplayNames(filters.display);
-    load(filters.identifiers);
+        return load(Stock.filter[filterKey].formatHTTP(true));
+      })
+      .catch(angular.noop);
```

Comment:
We can go ahead and remove this function in this refactor.  There is no need for the `angular.noop` - the `load()` function has error handling already.